flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyBufferPool.java

```diff
@@ -61,7 +60,9 @@ public NettyBufferPool(int numberOfArenas) {
 		checkArgument(numberOfArenas >= 1, "Number of arenas");
 		this.numberOfArenas = numberOfArenas;
 
-		if (!PlatformDependent.hasUnsafe()) {
+		try {
+			Class.forName("sun.misc.Unsafe");
+		} catch (ClassNotFoundException e) {
```

Comment:
After checking with @StefanRRichter, it seems this check is unnecessary anyway since `org.apache.flink.core.memory.MemoryUtils#UNSAFE` is getting the class if needed and if not existing, Flink should not start anyway.